internal/parser/reader_parser.go

```diff
@@ -5,7 +5,7 @@ import (
 	"io"
 )
 
-var splitFunc bufio.SplitFunc = func(data []byte, _ bool) (advance int, token []byte, err error) {
+var splitFunc bufio.SplitFunc = func(data []byte, atEOF bool) (advance int, token []byte, err error) {
 	if len(data) == 0 {
 		return
```

Comment:
Also, while we're here, can we get rid of naked returns? There is also one at the end of the function, which should be `return advance, token, nil`.
```suggestion
		return 0, nil, nil
```

Reply:
Agreed, will remove the naked returns from this function